### PR TITLE
Replacing the _getIndex method by UnderscoreJS method _.findIndex()

### PR DESCRIPTION
**Pull request by anuprulez**
- Reasons

The current 'getIndex' method returns the index of the specified value if present in the list/ array. This method can be replaced by one line code (a method of Underscore JS method) '_.findIndex' (UnderscoreJS methods are compactly written and faster). This method also takes care of the case when the queried value is not found in the list (returns -1 if value is not found. So no need to write one extra line for returning -1). 
- Changes

Previous 

``` javascript
        // search index
        for (var key in this.select_data) {
            if (this.select_data[key].id == value) {
                return key;
            }
        }
        // not found
        return -1;
```

Committed:

``` javascript
_.findIndex(this.select_data, {id: value});
```
- Tests

The test cases would remain the same as the values returned by the method remains the same.

### CODE REVIEW

File: client/galaxy/scripts/mvc/ui/ui-select.js [hunk 1]
```diff
@@ -181,14 +181,18 @@ var View = Backbone.View.extend(
...
-            if (this.select_data[key].id == value) {
-                return key;
-            }
-        }
+        // replaced the index search code 
+        // by UnderscoreJS method to make it faster and compact
+        _.findIndex(this.select_data, {id: value});
         
+	// Old code
+        //for (var key in this.select_data) {
```
Comment (bgruening) [inline client/galaxy/scripts/mvc/ui/ui-select.js]: The old code is not needed. Moreover you need to run `make client` from your GALAXY_ROOT to build the js libs.
Thanks @anuprulez !

**Comment by anuprulez** [comment]
For this commit, I have used an Underscore JS method '_findIndex' which is available in version **1.8.0**
http://underscorejs.org/#findIndex but in the Galaxy (https://github.com/galaxyproject/galaxy/blob/dev/client/galaxy/scripts/libs/underscore.js) we are using **1.7.0**. So the build is failing as that is method is not defined in the older version. 

**Work In Progress. Please do not merge.**

**Comment by anuprulez** [comment]
**Work In Progress. Please do not merge.**

**Comment by bgruening** [comment]
@anuprulez please have a look at this one: https://github.com/galaxyproject/galaxy/pull/2328

**Comment by anuprulez** [comment]
@bgruening: I have committed a change in bower.json file for downloading the latest version (1.8.3) of UnderscoreJS.

**Comment by bgruening** [comment]
@carlfeberhard /@guerler does this looks good?

**Comment by martenson** [comment]
merged manually, thanks for the contribution @anuprulez !

**Comment by bgruening** [comment]
Awesome! Thanks @anuprulez @martenson!

**Pull request merged**